llvm/test/MC/Disassembler/RISCV/c_slli.txt

```diff
@@ -0,0 +1,3119 @@
+# This test verifies the disassembler's behavior for all C.LUI opcodes with the
+# exception of those assigned to C.ADDI16SP
+#
+# RUN: llvm-mc --disassemble -triple=riscv32 -mattr=+c -M no-aliases \
+# RUN:     --show-encoding < %s | FileCheck --check-prefixes=GOOD %s
+#
+# RUN: llvm-mc --disassemble -triple=riscv64 -mattr=+c -M no-aliases \
+# RUN:     --show-encoding < %s | FileCheck --check-prefixes=GOOD,GOOD64 %s
+#
+# RUN: llvm-mc --disassemble -triple=riscv32 -mattr=+c -M no-aliases \
+# RUN:     --show-encoding < %s 2>&1 | FileCheck --check-prefix=BAD32 %s
+#
+# RUN: llvm-mc --disassemble -triple=riscv32  -mattr=+c -mattr=+no-rvc-hints \
+# RUN:     -M no-aliases --show-encoding < %s 2>&1 | \
+# RUN:   FileCheck --check-prefix=NOHINTS %s
+
+0x06 0x00 # GOOD: c.slli zero, 1
```

Comment:
This test is actually buggy. Each hex pattern should only appear once. All 4 command lines will process each hex value not just the ones with their check-prefix.

So the total number of "invalid instruction encoding" that appear in the output is more than how many we are checking for. The same bug is in the c.lui test and it's causing us to miss that we incorrectly decode 'c.lui zero, 0'.

Reply:
(PR #133921)